alacritty/src/input/mod.rs

```diff
@@ -278,6 +280,28 @@ impl<T: EventListener> Execute<T> for Action {
             },
             Action::Vi(ViAction::InlineSearchNext) => ctx.inline_search_next(),
             Action::Vi(ViAction::InlineSearchPrevious) => ctx.inline_search_previous(),
+            Action::Vi(ViAction::SemanticSearchForward) => {
+                let seed_text = match ctx.terminal().selection_to_string() {
+                    Some(selection) if !selection.is_empty() => selection,
+                    // Get semantic word at the vi cursor position.
+                    _ => ctx.semantic_word(ctx.terminal().vi_mode_cursor.point),
+                };
+
+                if !seed_text.is_empty() {
+                    ctx.start_seeded_search(Direction::Right, seed_text);
+                }
+            },
+            Action::Vi(ViAction::SemanticSearchBackward) => {
+                let seed_text = match ctx.terminal().selection_to_string() {
+                    Some(selection) if !selection.is_empty() => selection,
+                    // Get semantic word at the vi cursor position.
+                    _ => ctx.semantic_word(ctx.terminal().vi_mode_cursor.point),
+                };
+
+                if !seed_text.is_empty() {
+                    ctx.start_seeded_search(Direction::Left, seed_text);
+                }
+            },
```

Comment:
can you merge both and then dispatch the right function within based off the bounded `ViAction`?

Reply:
I guess I could, but that would require matching twice which is also not ideal.